### PR TITLE
Dat 3456, remove templates classified as navbox from articles on mercury

### PR DESCRIPTION
**Pull request by RafLeszczynski**
https://wikia-inc.atlassian.net/browse/DAT-3456

### CODE REVIEW

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 1]
```diff
@@ -257,10 +259,11 @@ public static function onShowEditLink( Parser &$this, &$showEditLink ) {
 	 * @param $report
 	 * @return bool
 	 */
-	public static function reportLimits( $parser, &$report ) {
+	public static function reportLimits( $parser, &$report )
+	{
```
Comment (dianafa) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: why? :(

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 2]
```diff
@@ -257,10 +259,11 @@ public static function onShowEditLink( Parser &$this, &$showEditLink ) {
...
 		global $wgArticleAsJson;
 
-		if ( $wgArticleAsJson ) {
+		if ($wgArticleAsJson) {
```
Comment (dianafa) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: why? :(

File: includes/wikia/services/TemplateClassificationService.class.php [hunk 3]
```diff
@@ -164,7 +165,12 @@ private function createApiClient() {
 		global $wgConsulUrl, $wgConsulServiceTag;
 		$urlProvider = new ConsulUrlProvider( $wgConsulUrl, $wgConsulServiceTag );
 		$apiProvider = new ApiProvider( $urlProvider );
-		return $apiProvider->getApi( self::SERVICE_NAME, TCSApi::class );
+		$apiClient = $apiProvider->getApi( self::SERVICE_NAME, TCSApi::class );
+
+		// overwrite curl timeout
```
Comment (dianafa) [inline includes/wikia/services/TemplateClassificationService.class.php]: let's explain _why_, not _what_ we are doing :)

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 4]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+use Swagger\Client\ApiException;
```
Comment (sebastianmarzjan) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: Per @dianafa commenting below, it looks like no changes in this file are needed

Comment (dianafa) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: hah, true that, so then:

why? :(

File: includes/wikia/DefaultSettings.php [hunk 5]
```diff
@@ -1830,3 +1830,14 @@
  * Reject attempts to fall back to the MediaWiki session for authentication.
  */
 $wgRejectAuthenticationFallback = true;
+
+/**
+ * Return template type for given template from Template Classification Service
+ */
+$wgAutoloadClasses['ExternalTemplateTypesProvider'] = "{$IP}/includes/wikia/parser/ExternalTemplateTypesProvider.class.php";
```
Comment (sebastianmarzjan) [inline includes/wikia/DefaultSettings.php]: You may want to consider moving the code to a subdir (parser/templatetypes/ ) and setting up autoload and hooks in a single file referenced from here

File: includes/wikia/parser/TemplateTypesParser.class.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+	 * @param string $text - template content
+	 * @param Title $finalTitle - template title object
+	 * @return bool
+	 */
+	public static function onFetchTemplateAndTitle( &$text, &$finalTitle ) {
+		global $wgEnableTemplateTypesParsing, $wgArticleAsJson, $wgCityId;
+
+		wfProfileIn( __METHOD__ );
+
+		if ( $wgArticleAsJson && $wgEnableTemplateTypesParsing ) {
```
Comment (sebastianmarzjan) [inline includes/wikia/parser/TemplateTypesParser.class.php]: I know it's a nitpick, though you may want to reverse the order ;-)

**Comment by sebastianmarzjan** [comment]
Left a few comments, I would also suggest moving dataproviders to the end of test code and leaving the real test cases at the top of the test class

File: includes/wikia/services/TemplateClassificationService.class.php [hunk 7]
```diff
@@ -164,7 +165,13 @@ private function createApiClient() {
 		global $wgConsulUrl, $wgConsulServiceTag;
 		$urlProvider = new ConsulUrlProvider( $wgConsulUrl, $wgConsulServiceTag );
 		$apiProvider = new ApiProvider( $urlProvider );
-		return $apiProvider->getApi( self::SERVICE_NAME, TCSApi::class );
+		$apiClient = $apiProvider->getApi( self::SERVICE_NAME, TCSApi::class );
+
+		// default curl timeout for API client is set to 0 which = to no timeout.
+		// Overwriting to minimal value which is 1.
+		$apiClient->getConfig()->setCurlTimeout(1);
```
Comment (sebastianmarzjan) [inline includes/wikia/services/TemplateClassificationService.class.php]: Just for reference, this is the timeout for entire operation, including connection.

File: includes/wikia/services/TemplateClassificationService.class.php [hunk 8]
```diff
@@ -164,7 +165,13 @@ private function createApiClient() {
 		global $wgConsulUrl, $wgConsulServiceTag;
 		$urlProvider = new ConsulUrlProvider( $wgConsulUrl, $wgConsulServiceTag );
 		$apiProvider = new ApiProvider( $urlProvider );
-		return $apiProvider->getApi( self::SERVICE_NAME, TCSApi::class );
+		$apiClient = $apiProvider->getApi( self::SERVICE_NAME, TCSApi::class );
+
+		// default curl timeout for API client is set to 0 which = to no timeout.
```
Comment (sebastianmarzjan) [inline includes/wikia/services/TemplateClassificationService.class.php]: Maybe it's worth to add what this means for php_curl

**Comment by sebastianmarzjan** [comment]
LGTM

**Comment by RafLeszczynski** [comment]
@SebastianMarzjan please take a look - last sanity check :-)

**Comment by RafLeszczynski** [comment]
@lgarczewski @lukaszkonieczny @kamilkoterba @adamkarminski pinging - knowledge sharing

**Comment by sebastianmarzjan** [comment]
:shipit:

File: includes/wikia/parser/templatetypes/TemplateTypes.setup.php [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+// Return template type for given template from Template Classification Service
```
Comment (kamilkoterba) [inline includes/wikia/parser/templatetypes/TemplateTypes.setup.php:3]: Can we have this explanation in class documentation as well?

File: includes/wikia/parser/templatetypes/TemplateTypes.setup.php [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+$wgAutoloadClasses['ExternalTemplateTypesProvider'] = __DIR__ . '/ExternalTemplateTypesProvider.class.php';
+
+// Handles special parsing of templates classified by Template Classification Service
```
Comment (kamilkoterba) [inline includes/wikia/parser/templatetypes/TemplateTypes.setup.php:6]: Same as above

**Comment by adamkarminski** [comment]
LGTM

**Pull request merged**